jina/peapods/runtimes/zmq/zed.py

```diff
@@ -389,10 +389,12 @@ def _get_docs(self, field: str) -> 'DocumentArray':
         else:
             result = getattr(self.request, field)
 
-        # to unify all length=0 DocumentArray (or any other results) will simply considered as None
-        # otherwise the executor has to handle DocArray(0)
-        if len(result):
-            return result
+        # to unify all length=0 DocumentArray (or any other results) will simply considered as []
+        # otherwise the executor has to handle None values
+
+        if result is None:
```

Comment:
I would do it on client side. it is good to have the option to have docs as None (as Dump) or whatever, and it is what someone would expect if None is in inpus

Reply:
Otherwise the condition of empty batch vs on purpose None inputs cannot be separated

Reply:
I what file would you put it?

Reply:
Look for this function:

```python
U need to fix most likely this function:
 def _add_docs_groundtruths(req, batch, data_type, _kwargs):
     for content in batch:
         if isinstance(content, tuple) and len(content) == 2:
             # content comes in pair,  will take the first as the input and the second as the ground truth
 
             # note how data_type is cached
             d, data_type = _new_doc_from_data(content[0], data_type, **_kwargs)
             gt, _ = _new_doc_from_data(content[1], data_type, **_kwargs)
             req.docs.append(d)
             req.groundtruths.append(gt)
         else:
             d, data_type = _new_doc_from_data(content, data_type, **_kwargs)
             req.docs.append(d)
```

Reply:
I think u need to consider when batch is empty

Reply:
not 100% sure

Reply:
I think the logic of adding the groundtruth should be separated from `None` or [] decisions

Reply:
Also it poses the problem that is cheating the user. This new DocArray p

Reply:
maybe instruct users to do `bool(docArray)` to check if the document array is empty ?